lib/src/intl_suggestors/utils.dart

```diff
@@ -8,15 +8,25 @@ import 'package:over_react_codemod/src/util/element_type_helpers.dart';
 
 import 'constants.dart';
 
+var result;
 // -- Node Validation --
+
+/// For a string interpolation, get all the non-interpolated parts in a string,
+/// separated by spaces.
+String textFromInterpolation(StringInterpolation body) => body.elements
+      .whereType<InterpolationString>()
+      .map((each) => each.value)
+      .join(' ')
+      .trim();
 bool isValidStringInterpolationNode(AstNode node) {
   if (node is! StringInterpolation) return false;
   //We do not need to localize single values.  This should be handled by the
   // variable being passed in.  IE: ..foo = '$bar' becomes [', '$bar', ']
   if (node.elements.length == 3 &&
       node.elements.first.toString() == node.elements.last.toString())
     return false;
-  return true;
+   result=textFromInterpolation(node);
+  return result.isNotEmpty && result.contains(RegExp('[a-zA-Z]'));
```

Comment:
The thing I was suggesting to pull out into a top-level variable was the RegExp, not the result variable. e.g.

RegExp alphabetMatcher = RegExp('[a-zA-Z]');

Reply:
Code updated as per your sugesstion